src/Entities/StrawberryJam/DashZipMover.cs

```diff
@@ -0,0 +1,411 @@
+                Vector2 collisionPoint = dir != 1 ? TopLeft : TopRight;
+
+                int particleOffset = 4;
+                if (dir == 1)
+                    particleOffset = Math.Min((int) Width - 12, 20);
+
+                int particleWidth = (int) Width;
+                if (dir == -1)
+                    particleWidth = Math.Max(16, (int) Width - 16);
+
+                if (Scene.CollideCheck<Solid>(collisionPoint + new Vector2(dir * -2, -2f)))
+                    for (int x = particleOffset; x < particleWidth; x += 8)
+                        SceneAs<Level>().ParticlesFG.Emit(ZipMover.P_Scrape, TopLeft + new Vector2(x + dir * 2f, -1f), dir == 1 ? (float) Math.PI * 3f / 4f : (float) Math.PI / 4f);
+
+                if (Scene.CollideCheck<Solid>(collisionPoint + new Vector2(dir * -2, Height + 2f)))
+                    for (int x = particleOffset; x < particleWidth; x += 8)
+                        SceneAs<Level>().ParticlesFG.Emit(ZipMover.P_Scrape, BottomLeft + new Vector2(x + dir * 2f, 0f), dir == 1 ? (float) Math.PI * -3f / 4f : -(float) Math.PI / 4f);
+            }
+        }
+
+        private IEnumerator Sequence()
+        {
+            Vector2 start = Position;
+
+            while (true)
+            {
+                if (!triggered)
+                {
+                    yield return null;
+                    continue;
+                }
+
+                sfx.Play(game_dash_zip_mover_zip_mover);
+
+                Input.Rumble(RumbleStrength.Medium, RumbleLength.Short);
+                StartShaking(0.1f);
+                yield return 0.1f;
+
+                streetlight.SetAnimationFrame(3);
+                StopPlayerRunIntoAnimation = false;
+
+                float at2 = 0f;
+
+                while (at2 < 1f)
+                {
+                    yield return null;
+                    at2 = Calc.Approach(at2, 1f, 2f * Engine.DeltaTime);
+                    percent = Ease.SineIn(at2);
+                    Vector2 vector = Vector2.Lerp(start, target, percent);
+                    ScrapeParticlesCheck(vector);
+                    if (Scene.OnInterval(0.1f))
+                        pathRenderer.CreateSparks();
+                    MoveTo(vector);
+                }
+
+                StartShaking(0.2f);
+                Input.Rumble(RumbleStrength.Strong, RumbleLength.Medium);
+                streetlight.SetAnimationFrame(2);
+                SceneAs<Level>().Shake();
+                StopPlayerRunIntoAnimation = true;
+                yield return 0.5f;
+
+                StopPlayerRunIntoAnimation = false;
+                streetlight.SetAnimationFrame(1);
+                triggered = false;
+                target = start;
+                start = Position;
+                sfx.Stop();
+            }
+        }
+
+        // Works with negatives, might be useful to move this in some util class?
+        private float Mod(float x, float m)
+        {
+            return (x % m + m) % m;
+        }
```

Comment:
I don't understand what does this function computes😅

Reply:
This is a positive modulo operator, which is notably differentiable @ 0. Positives are normal, and negatives follow the pattern of 'add m to x until it is nonnegative, then return that value'
-4 % 3 normally equals -1, with this mod function it returns 2 (-4 + 3 + 3 = 2)

Reply:
I see, mb rename this function `PositiveModulo` to make this clear?
```suggestion
        // Works with negatives, might be useful to move this in some util class?
        private float PositiveModulo(float x, float m)
        {
            return (x % m + m) % m;
        }
```

Reply:
i bet this exact method is defined at least a couple times in the projet. check if it isn't already in `Util`, and if not, maybe it can be refactored there.